Implement Discord-style user pill, adjust sidebars and main layout spacing

GID0317 opened the pull request:
This PR implements new Discord-style floating user profile panel at the bottom of the left sidebar, featuring:
- Add Discord-style floating user pill at bottom of left sidebar
- Spans both server list and channel list columns
- Includes user avatar, display name, username, and settings button
- Multi-theme support with acrylic backgrounds (SunValley (For floating user profile) /Fluent) and solid fallback (Performance)
- Added bottom padding to server/channel/DM lists to prevent overlap

Tested on: Windows 11 Version 25H2 (OS Build 26220.7344) x64

So far, it works well on my devices! and both the Fluent and Performance themes are supported. 
Note: Font was using latest SegoeFluentIcons (This part isn’t included in this PR and will be submitted in another one.)

Discord-style floating user profile:
<img width="1247" height="698" alt="Screenshot 2025-12-17 010914" src="https://github.com/user-attachments/assets/9ad42394-85a5-40f0-89c3-0b5d3ee24a78" />
<img width="1247" height="698" alt="image" src="https://github.com/user-attachments/assets/f7642b88-d551-49a7-b905-908358367c53" />
Attempts to scroll to bottom:
<img width="1382" height="741" alt="image" src="https://github.com/user-attachments/assets/7a99cace-0842-49af-a9ba-43024f0b8bdc" />
Update:
<img width="368" height="105" alt="image" src="https://github.com/user-attachments/assets/f23940f7-0fe7-4744-ae31-4d49b150c1df" />



Spacing between layout:
<img width="1920" height="1020" alt="image" src="https://github.com/user-attachments/assets/6da2582d-4e3e-4d62-b6e8-65ebce413f81" />

GuildListItem:
<img width="1920" height="1020" alt="image" src="https://github.com/user-attachments/assets/8f3918be-945e-4d44-a13f-2a582c74dc37" />

GID0317 committed:
Add floating user panel pill and theme resources

Adds a Discord-style floating user panel pill to the left sidebar (DiscordPage.xaml).
Introduces theme resources for the pill (DiscordPage_UserPill_Background, DiscordPage_UserPill_CornerRadius) in SunValley, Fluent and Performance.
Adds bottom padding to DM/guild lists to prevent overlap with the pill.
Fix asymmetrical guild item view list
Improve the spacing between layout to  use 4 instead of 8

GID0317 committed:
Make bottom left and right DM/Channel list to not use rounded corners

WamWooWam commented on Unicord.Universal/Pages/DiscordPage.xaml:
```diff
@@ -224,37 +227,49 @@
+                        <TextBlock Text="{Binding CurrentUser.DisplayName}" FontSize="13" FontWeight="SemiBold" TextTrimming="CharacterEllipsis"/>
+                        <TextBlock Foreground="{ThemeResource ApplicationSecondaryForegroundThemeBrush}" 
+                                   FontSize="11" 
+                                   FontFamily="Segoe UI Variable Small" 
```
Segoe UI Variable only exists as a font on Windows 11, I'd suggest pulling that out into a theme resource and using regular Segoe UI on other devices

WamWooWam replied:
Oh wait _I_ did this, no better time like the present to fix it I guess 😅

WamWooWam commented on Unicord.Universal/Pages/DiscordPage.xaml:
```diff
@@ -224,37 +227,49 @@
+                      Padding="8,6"
+                      Height="52">
+                    <Grid.Shadow>
+                        <ThemeShadow />
```
ThemeShadow doesn't exist on Windows 10 prior to 1903, so you'll need to version prefix this. 

Add the following XML namespace to the top of the file
```xml
xmlns:w1903="http://schemas.microsoft.com/winfx/2006/xaml/presentation?IsApiContractPresent(Windows.Foundation.UniversalApiContract,8)">
```

then prefix it, e.g.
```xml
<w1903:Grid.Shadow>
    <w1903:ThemeShadow />
</w1903:Grid.Shadow>
```

WamWooWam reviewed (changes requested):
Really minor things, once again nice work

GID0317 committed:
moved Font family to theme resource

Created DiscordPage_UserPill_SecondaryFontFamily resource:
SunValley (Windows 11): Segoe UI Variable Small
Fluent (Windows 10): Segoe UI
Performance (Windows 10): Segoe UI

GID0317 committed:
Update DiscordPage.xaml

dded xmlns:w1903 namespace and prefixed Grid.Shadow and ThemeShadow with w1903: so it only applies on Windows 10 1903+
Improve margin of profile to be more consistent
make font use Font family theme resource
